src/main/java/fr/quatrevieux/araknemu/data/living/repository/implementation/sql/SqlPlayerRepository.java

```diff
@@ -169,7 +178,7 @@ public Collection<Player> findByAccount(int accountId, int serverId) {
     @Override
     public boolean nameExists(int serverId, String name) {
         return utils.aggregate(
-            "SELECT COUNT(*) FROM PLAYER WHERE PLAYER_NAME = ? AND SERVER_ID = ?",
+            "SELECT COUNT(*) FROM PLAYER WHERE PLAYER_NAME = ? AND SERVER_ID = ? AND DELETED_AT IS NULL",
```

Comment:
Delete at should not be filtered here, because of the unique index on server id and name.